src/D2L.Bmx/BmxConfigProvider.cs

```diff
@@ -0,0 +1,84 @@
+using System.Security;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.FileProviders.Physical;
+namespace D2L.Bmx;
+
+internal interface IBmxConfigProvider {
+	BmxConfig GetConfiguration();
+}
+
+internal class BmxConfigProvider : IBmxConfigProvider {
+
+	public BmxConfig GetConfiguration() {
+		// Main config is at ~/.bmx/config
+		var configLocation = Path.Join( Environment.GetFolderPath( Environment.SpecialFolder.UserProfile ), ".bmx",
+			"config" );
+
+		// If set, we recursively look up from CWD (all the way to root) for additional bmx config files (labelled as .bmx)
+		var configBuilder = new ConfigurationBuilder().AddIniFile( configLocation, optional: true );
+
+		bool.TryParse( configBuilder.Build()["allow_project_configs"], out bool shouldReadProjectConfig );
+
+		string? projectConfigPath = null;
+
+		if( shouldReadProjectConfig ) {
+			projectConfigPath = GetProjectConfigPath();
+		}
+
+		if( shouldReadProjectConfig && !string.IsNullOrEmpty( projectConfigPath ) ) {
+			// Default file provider ignores files prefixed with ".", we need to provide our own as a result
+			var fileProvider =
+				new PhysicalFileProvider( Path.GetDirectoryName( projectConfigPath )!, ExclusionFilters.None );
+			configBuilder.AddIniFile( fileProvider, Path.GetFileName( projectConfigPath ), optional: false,
```

Comment:
why is it necessary to do this here but not at [L19](https://github.com/Brightspace/bmx/pull/234/files#diff-2eb5576e543a9b7cb2ee5a009fd10a20e4bbeb2a3115370f6d75de9403f7deb1R19)?

Reply:
Do you mean the bit about passing a different fileProvider? Its to handle reading project configs that are in ``.bmx``files, since those are ignored by the Configuration Builder as hidden files.

Reply:
yes, but at L19 aren't we also reading a ".bmx" file? why is it fine to not pass a custom configured file provider there?

Reply:
L19 is reading the ``~/.bmx/config`` file

Reply:
Ok so from my understanding L19 doesn't need to be changed then. Should the `allow_project_configs` logic just be removed now or does it require more discussion?

Reply:
so... the difference is between .bmx _folder_ & .bmx _file_?
i.e. the default file provider includes folders starting with a dot, but not files starting with a dot?

Reply:
> Should the allow_project_configs logic just be removed now or does it require more discussion?

Sorry I meant in the future, for now we should keep this for 1:1 with current BMX

> i.e. the default file provider includes folders starting with a dot, but not files starting with a dot?

yes

Reply:
Let's keep the `allow_project_configs` config for now. I know it's used by some to disable project config lookup to resolve some infinite recursion issue in the Go version on Linux/Mac machines. The current code here could suffer from the same problem. I wouldn't remove it unless we can be sure people won't be upset without it.